docs/design/build-system.md

```diff
@@ -149,14 +149,32 @@ Aside from tree-shaking, scripts and styles (as defined in the sources above) al
 
 There are some steps that are meant to operate over existing whole applications and thus happen after the webpack compilation finishes and outputs files.
 
-The fist step is differential loading, where we take code that is targeting modern browsers and produces from it a compatible version for older browsers.
-`index.html` is then modified in such a way that both modern and older browsers only load their corresponding scripts.
+### Differential Loading
+Differential loading is a strategy that allows your web application to support multiple browsers, but only load the necessary code that the browser needs. When differential loading is enabled, the CLI generates two separate sets of application bundles.
+
+* The first contains ES2015 syntax, takes advantage of built-in support in modern browsers, ships fewer polyfills, and results in a smaller total size.
+* The second contains code in the older ES5 syntax, along with all necessary polyfills for Angular to function. This results in a larger total size, but supports older browsers.
+
+To generate these two sets, CLI uses a bundle level transformation process.
+This process requires that only one full compilation of the application in ES2015 syntax to take place.
+A copy of the output files are then transformed (commonly referred to as down-leveled) to ES5 compatible syntax. ES2015+ syntax elements such as classes are converted into functionally equivalent ES5 code structures. An additional ES5-only polyfills file is also generated that contains the required Angular polyfills for ES5-only browsers.
+This process also guarantees that the entire bundled application is ES5 compliant including third-party code.
+
+To support loading the file sets in the appropriate browsers, the HTML `script` element's `type` and `nomodule` attributes are leveraged.  Browsers will only load a script with a known type.  The ES2015 files are referenced
```

Comment:
Nit: odd like breaks and double spaces in this paragraph. Would be a bit cleaner if sentences were in different lines.